cromwell_tools/_cromwell_api.py

```diff
@@ -0,0 +1,425 @@
+        if 'dependencies_json' in manifest:
+            files['workflowDependencies'] = manifest['dependencies_json']
+        if 'options_json' in manifest:
+            files['workflowOptions'] = manifest['options_json']
+
+        if label:
+            files['labels'] = label
+        if collection_name:  # TODO: this check should happen elsewhere, related to caas
+            files['collectionName'] = collection_name
+
+        response = cls.requests.post(url=auth.url + cls._workflow_endpoint,
+                                     files=files,
+                                     auth=auth.auth,
+                                     headers=auth.header)
+
+        response.raise_for_status()
+        return response
+
+    @classmethod
+    def wait(cls, workflow_ids, timeout_minutes, auth, poll_interval_seconds=30, verbose=True):
+        """Wait until cromwell returns successfully for each provided workflow
+
+        Given a list of workflow ids, wait until cromwell returns successfully for each status, or
+        one of the workflows fails or is aborted.
+
+        Args:
+        workflow_ids (List): Workflow ids to wait for terminal status
+        timeout_minutes (int): Maximum number of minutes to wait
+        auth (cromwell_tools._cromwell_auth.CromwellAuth): Authentication class holding headers
+            or auth information to a Cromwell server
+        poll_interval_seconds (Optional[int]): Number of seconds between checks for workflow
+            completion (default 30)
+        verbose (Optional[bool]): If True, report to stdout when all workflows succeed
+            (default True)
+        """
+        start = datetime.now()
+        timeout = timedelta(minutes=int(timeout_minutes))
+
+        while True:
+
+            if datetime.now() - start > timeout:
+                msg = 'Unfinished workflows after {0} minutes.'
+                raise Exception(msg.format(timeout))
+
+            all_succeeded = True
+            for uuid in workflow_ids:
+                response = cls.status(uuid, auth)
+                status = cls._parse_status(response)
+                if status in cls._failed_statuses:
+                    raise WorkflowFailedException('Workflow {0} returned status {1}'.format(
+                            uuid, status))
+                elif status != 'Succeeded':
+                    all_succeeded = False
+
+            if all_succeeded:
+                if verbose:
+                    print('All workflows succeeded!')
+                return
+
+            time.sleep(poll_interval_seconds)
+
+    @classmethod
+    def release_hold(cls, uuid, auth, raise_for_status=False):
+        """Request Cromwell to release the hold on a workflow.
+
+        It will switch the status of a workflow from ‘On Hold’ to ‘Submitted’ so it can be picked for running. For
+        a workflow that was not submitted with `workflowOnHold = true`, Cromwell will ignore the request.
```

Comment:
> "For a workflow that was not submitted with `workflowOnHold = true`, Cromwell will ignore the request."

Is this still the intended behavior for this function? Cromwell will throw a 403 error if the workflow is not on hold, so we would have to handle that error here.

Reply:
Great catch, this docstring is lying, the post request will raise 403 and, this function will return the errored response or raise the error to the users(if raise_for_status==True).